client/src/components/History/Content/ExpandedItems.js

```diff
@@ -22,7 +22,10 @@ export default {
         },
         setExpanded(item, val) {
             const key = this.getItemKey(item);
-            const newSet = new Set(this.items);
+            // disable expansion of multiple items for now, expanded items
+            // are kept up-to-date and there should be at least naive caching
+            // in place in order to avoid piling url requests
+            const newSet = new Set();
```

Comment:
That is severely limiting in how you can use the history. What do you mean by piling url requests ? Could you listen for history updates and update store items ?

Reply:
Had a discussion about this, so the idea here is just that we need some basic caching here, and yes I do think that history updates from the store could work but I have not focussed on it yet. I will remove this commit for now although this leaves us open to performance issues if e.g. a user expands dozens of datasets and keeps scrolling very fast up and down. Note: Only visible expanded items issue an update request.